[server] add wl_client_get_credentials (#398)
Smithay/wayland-rs: CHANGELOG.md

## Unreleased

> * [server] Allow the `Credentials` to be retrieved from a client socket (implements `wl_client_get_credentials`).

Reviewer: ```suggestion
* [server] Introduce `Client::get_credentials()` to allow retrieving the credentials of client socket (implements `wl_client_get_credentials`).
```

Author: Yeah, that is way better, thanks!
